vowpalwabbit/parse_args.cc

```diff
@@ -557,6 +557,13 @@ const char* are_features_compatible(vw& vw1, vw& vw2)
 
   return nullptr;
 }
+
+// Compares the header against CB, CCB and CS header formats.
+bool ec_is_example_header(example& ec)
+{
+  return CB::ec_is_example_header(ec) || COST_SENSITIVE::ec_is_example_header(ec) || CCB::ec_is_example_header(ec);
```

Comment:
This is horrifically unsafe. CB and CS are reasonably safe because have exactly the same in-memory layout. However, CCB is totally different, which means this check is undefined behavior

Reply:
You're right, it's totally unsafe. What do you think we should do to fix this properly?